include/ginkgo/core/base/math.hpp

```diff
@@ -102,6 +103,21 @@ using std::sqrt;
 namespace detail {
 
 
+/** Returns an unsigned type matching the size of the given float type. */
+template <typename T>
+struct float_to_bytes_impl {};
+
+template <>
+struct float_to_bytes_impl<double> {
+    using type = uint64;
+};
+
+template <>
+struct float_to_bytes_impl<float> {
+    using type = uint32;
+};
```

Comment:
Please name it `float_to_uint_impl` because the type you are using is not `byte`.